addon/templates/components/ember-tbody/ember-tbody.hbs

```diff
@@ -24,14 +24,24 @@ as |rowValue|
         checkboxSelectionMode=checkboxSelectionMode
     as |api|
     }}
-        {{#if hasBlock}}
+        {{#if @tableMeta.expandedRowComponent}}
             {{yield
                 (hash
                     rowValue=api.rowValue
                     rowMeta=api.rowMeta
                     cells=api.cells
                     rowSelectionMode=api.rowSelectionMode
-                    row=(component "ember-tr" api=api)
+                    row=(component "row/table-row" api=api tableMeta=@tableMeta)
+                )
+            }}
+        {{else if hasBlock}}
+            {{yield
+                (hash
+                    rowValue=api.rowValue
+                    rowMeta=api.rowMeta
+                    cells=api.cells
+                    rowSelectionMode=api.rowSelectionMode
+                    row=(component "ember-tr" api=api tableMeta=@tableMeta)
```

Comment:
So my thought was we would keep this logic more generic, to accept any type of custom row component, instead of just looking for the "expanded row" stuff. So here, the changes to this block would be much simpler, like this:

```hbs
{{#if hasBlock}}
    {{yield
        (hash
            rowValue=api.rowValue
            rowMeta=api.rowMeta
            cells=api.cells
            rowSelectionMode=api.rowSelectionMode
            row=(component @rowComponent 
                api=api 
                tableMeta=@tableMeta
            )
        )
    }}
{{else}}
    {{ember-tr api=api}}
{{/if}}
```

Where `@rowComponent` is an arg sent into the ember-tbody from the Table component, who itself also has it defined as a root arg (not in `@tableMeta`), with an default of the existing `ember-tr` component:
```ts
//in Table.ts
@argDefault rowComponent = 'ember-tr';
```

Then, the app would supply what row component they want to use like:
```
<Table 
    @rowComponent="row/expandable-row"
    @tableMeta={{hash
        expandedRowComponent="table/row/my-custom-row-content"
    }}
/>
```
The `row/expandable-row` bit is the "third component" i was talking about before, which is another row component we define in the addon, which extends our new glimmerified `row/table-row`component, and adds the `isExpanded` logic to its component/template (which would be removed from where it is right now in `row/table-row`). That's why I mentioned the row/table-row component would never really be directly used, its purpose is mainly to serve as just a base glimmer component that other custom row components (both in the addon, and in apps) can extend from.

And lastly, because we chose to use the expandable-row custom row in this example, we have to tell that component what to use as its "expanded content" component, which we send along in the `tableMeta` object.